packages/lodestar/src/eth1/impl/util.ts

```diff
@@ -0,0 +1,20 @@
+import {IDepositEvent} from "../interface";
+
+/**
+ * Return deposit events of blocks grouped/sorted by block number and deposit index
+ * @param depositEvents range deposit events
+ */
+export function groupDepositEventsByBlock(rangeDepositEvents: IDepositEvent[]): Map<number, IDepositEvent[]> {
+  if (!rangeDepositEvents || rangeDepositEvents.length === 0) {
+    return new Map();
+  }
+  rangeDepositEvents.sort((event1, event2) => event1.index - event2.index);
+  return rangeDepositEvents.reduce<Map<number, IDepositEvent[]>>((previousValue, currentValue) => {
```

Comment:
Might be nice to call the variables `groupedEvents` and `event`, or something like that.